pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/url.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"net/url"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"knative.dev/pkg/apis"
+)
+
+type UrlTemplateValues struct {
```

Comment:
Is it ok to support values from `annotations` and `labels` as well, similar to [DomainTemplate](https://github.com/kserve/kserve/blob/39ec2d19de1d856a9527b800fda8fb380bc9441f/pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/domain.go#L30-L36).

Reply:
ACK! Let's decide whether we go with urlTemplate and if we do then I will add a fixup on top. Thanks!